artemis-selector/src/main/java/org/apache/activemq/artemis/selector/filter/JAXPXPathEvaluator.java

```diff
@@ -56,8 +60,11 @@ protected boolean evaluate(String text) {
 
    protected boolean evaluate(InputSource inputSource) {
       try {
-         return ((Boolean)xpath.evaluate(xpathExpression, builder.parse(inputSource), XPathConstants.BOOLEAN)).booleanValue();
+         synchronized (builder) {
```

Comment:
shouldn't there be multiple instances created instead?  I mean... it seems the factory always returns the same instance?

Reply:
I'm not exactly sure if there _should_ be multiple instances. The `builder` is created in a `static` block in `org.apache.activemq.artemis.selector.filter.XPathExpression`.